src/nemos/basis/_basis.py

```diff
@@ -506,15 +536,20 @@ def __pow__(self, exponent: int) -> MultiplicativeBasis:
             If the integer is zero or negative.
         """
         if not isinstance(exponent, int):
-            raise TypeError("Exponent should be an integer!")
+            raise TypeError("Basis exponent should be an integer!")
 
         if exponent <= 0:
-            raise ValueError("Exponent should be a non-negative integer!")
+            raise ValueError("Basis exponent should be a non-negative integer!")
+
+        # default case
+        if exponent == 1:
+            return deepcopy(self)
 
-        result = self
-        for _ in range(exponent - 1):
-            result = result * self
-        return result
+        mul = MultiplicativeBasis(self, self)
+        with mul._set_shallow_copy_temporarily(True):
+            for _ in range(2, exponent):
+                mul = MultiplicativeBasis(mul, deepcopy(self))
```

Comment:
```suggestion
                mul *= deepcopy(self)
```
optional, but I think it's more readable and roughly the same speed (whether it was faster or slower fluctuated from test to test)